crates/red_knot_test/src/parser.rs

```diff
@@ -328,6 +344,16 @@ impl<'s> Parser<'s> {
         // We never pop the implicit root section.
         let section = self.stack.top();
 
+        if captures.name("end").unwrap().is_empty() {
+            let code_block_start = self.cursor.token_len();
+            let row = self.source.count_lines(TextRange::up_to(code_block_start)) + 1;
+
+            return Err(anyhow::anyhow!(
+                "Unterminated code block at: {}:{row}:0",
+                self.source_path
```

Comment:
All errors are already output with the full path as a prefix, so as you have it this ends up rendering as:

```
Error parsing `/Users/carlmeyer/projects/ruff/crates/red_knot_python_semantic/resources/mdtest/scopes/unbound.md`: Unterminated code block at: /Users/carlmeyer/projects/ruff/crates/red_knot_python_semantic/resources/mdtest/scopes/unbound.md:47:0
```

Which seems a bit redundant :)

```suggestion
                "Unterminated code block starting on line {row}",
```